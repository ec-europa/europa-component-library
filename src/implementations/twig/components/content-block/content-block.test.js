import {
  merge,
  renderTwigFileAsNode,
  renderTwigFileAsHtml,
} from '@ecl/test-utils';
import { axe, toHaveNoViolations } from 'vitest-axe';

import dataImage from '@ecl/specs-component-content-block/demo/data';

expect.extend(toHaveNoViolations);

const dataSimpleTitle = { ...dataImage, title: 'title' };

describe('Content block', () => {
  const template = '@ecl/content-block/content-block.html.twig';
  const render = (params) => renderTwigFileAsNode(template, params);

  describe('Default', () => {
    test('renders correctly', () => {
      expect.assertions(1);
      return expect(render(dataImage)).resolves.toMatchSnapshot();
    });

    test('renders correctly with simple title', () => {
      expect.assertions(1);
      return expect(render(dataSimpleTitle)).resolves.toMatchSnapshot();
    });

    test('renders correctly with extra class names', () => {
      expect.assertions(1);

      const withExtraClasses = merge(dataImage, {
        extra_classes: 'custom-class custom-class--test',
      });

      return expect(render(withExtraClasses)).resolves.toMatchSnapshot();
    });

    test('renders correctly with extra attributes', () => {
      expect.assertions(1);

      const withExtraAttributes = merge(dataImage, {
        extra_attributes: [
          { name: 'data-test', value: 'data-test-value' },
          { name: 'data-test-1', value: 'data-test-value-1' },
        ],
      });

      return expect(render(withExtraAttributes)).resolves.toMatchSnapshot();
    });

    test('renders correctly with list extra classes', () => {
      expect.assertions(1);

      dataImage.lists.forEach((list) => {
        list.extra_classes = 'a-list-extra-class';
      });

      return expect(render(dataImage)).resolves.toMatchSnapshot();
    });

    test(`passes the accessibility tests`, async () => {
      expect(
        await axe(await renderTwigFileAsHtml(template, dataImage, true)),
      ).toHaveNoViolations();
    });
  });
});
