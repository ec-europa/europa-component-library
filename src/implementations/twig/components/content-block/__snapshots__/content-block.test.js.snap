// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Content block Default renders correctly 1`] = `
<jest>
  <div
    class="ecl-content-block"
    data-ecl-auto-init="ContentBlock"
    data-ecl-content-block=""
  >
    <ul
      class="ecl-content-block__label-container"
    >
      <li
        class="ecl-content-block__label-item"
      >
        <span
          class="ecl-label ecl-label--highlight"
        >
          highlight
        </span>
      </li>
      <li
        class="ecl-content-block__label-item"
      >
        <span
          class="ecl-label ecl-label--high"
        >
          high importance
        </span>
      </li>
    </ul>
    <ul
      class="ecl-content-block__primary-meta-container"
    >
      <li
        class="ecl-content-block__primary-meta-item"
      >
        PRIMARY META
      </li>
      <li
        class="ecl-content-block__primary-meta-item"
      >
        DD Month Year
      </li>
    </ul>
    <h1
      class="ecl-content-block__title"
      data-ecl-title-link=""
    >
      <a
        class="ecl-link ecl-link--standalone"
        href="/example"
      >
        Title
      </a>
    </h1>
    <div
      class="ecl-content-block__description"
    >
      Lorem ipsum dolor sit amet, consectetur adipiscing elit. Phasellus gravida ipsum ut lorem cursus, quis tincidunt sem viverra. Nunc vestibulum, mauris quis porta venenatis, justo odio commodo tellus
    </div>
    <ul
      class="ecl-content-block__secondary-meta-container"
    >
      <li
        class="ecl-content-block__secondary-meta-item"
      >
        <svg
          aria-hidden="true"
          class="ecl-icon ecl-icon--s ecl-content-block__secondary-meta-icon"
          focusable="false"
        >
          <use
            xlink:href="/icons.svg#calendar"
          />
        </svg>
        <span
          class="ecl-content-block__secondary-meta-label"
        >
          2018/10/22
        </span>
      </li>
      <li
        class="ecl-content-block__secondary-meta-item"
      >
        <svg
          aria-hidden="true"
          class="ecl-icon ecl-icon--s ecl-content-block__secondary-meta-icon"
          focusable="false"
        >
          <use
            xlink:href="/icons.svg#location"
          />
        </svg>
        <span
          class="ecl-content-block__secondary-meta-label"
        >
          Luxembourg
        </span>
      </li>
    </ul>
    <div
      class="ecl-content-block__list-container"
    >
      <dl
        class="ecl-description-list ecl-content-block__list"
      >
        <dt
          class="ecl-description-list__term"
        >
          Standalone links
        </dt>
        <dd
          class="ecl-description-list__definition ecl-description-list__definition--link"
        >
          <a
            class="ecl-link ecl-link--standalone ecl-description-list__definition-item"
            href="/example"
          >
            Lorem ipsum dolor sit amet
          </a>
          <a
            class="ecl-link ecl-link--standalone ecl-description-list__definition-item"
            href="/example"
          >
            Lorem ipsum dolor sit amet
          </a>
        </dd>
        <dt
          class="ecl-description-list__term"
        >
          Standard text
        </dt>
        <dd
          class="ecl-description-list__definition"
        >
          Lorem ipsum dolor sit amet, 
          <a
            class="ecl-link"
            href="/example"
          >
            consectetur adipiscing elit
          </a>
          .
        </dd>
        <dt
          class="ecl-description-list__term"
        >
          Links inline
        </dt>
        <dd
          class="ecl-description-list__definition ecl-description-list__definition--inline"
        >
          <a
            class="ecl-link ecl-link--standalone ecl-description-list__definition-item"
            href="/example"
          >
            Lorem ipsum dolor sit amet
          </a>
          <a
            class="ecl-link ecl-link--standalone ecl-description-list__definition-item"
            href="/example"
          >
            Lorem ipsum dolor sit amet
          </a>
        </dd>
        <dt
          class="ecl-description-list__term"
        >
          Taxonomy list
        </dt>
        <dd
          class="ecl-description-list__definition ecl-description-list__definition--taxonomy"
        >
          <span
            class="ecl-description-list__definition-item"
          >
            Taxonomy item 1
          </span>
          <span
            class="ecl-description-list__definition-item"
          >
            Taxonomy item 2
          </span>
          <span
            class="ecl-description-list__definition-item"
          >
            Taxonomy item 3
          </span>
        </dd>
      </dl>
    </div>
    <div
      class="ecl-content-block__link-container"
    >
      <ul
        class="ecl-content-block__link-list"
      >
        <li
          class="ecl-content-block__link-item"
        >
          <a
            class="ecl-link ecl-link--standalone"
            href="/example"
          >
            Primary link 1
          </a>
        </li>
        <li
          class="ecl-content-block__link-item"
        >
          <a
            class="ecl-link ecl-link--standalone"
            href="/example"
          >
            Primary link 2
          </a>
        </li>
        <li
          class="ecl-content-block__link-item"
        >
          <a
            class="ecl-link ecl-link--standalone"
            href="/example"
          >
            Primary link 3
          </a>
        </li>
        <li
          class="ecl-content-block__link-item"
        >
          <a
            class="ecl-link ecl-link--standalone"
            href="/example"
          >
            Primary link 4
          </a>
        </li>
      </ul>
      <ul
        class="ecl-content-block__link-list"
      >
        <li
          class="ecl-content-block__link-item"
        >
          <a
            class="ecl-link ecl-link--standalone"
            href="/example"
          >
            Secondary link 1
          </a>
        </li>
        <li
          class="ecl-content-block__link-item"
        >
          <a
            class="ecl-link ecl-link--standalone"
            href="/example"
          >
            Secondary link 2
          </a>
        </li>
      </ul>
    </div>
  </div>
</jest>
`;

exports[`Content block Default renders correctly with extra attributes 1`] = `
<jest>
  <div
    class="ecl-content-block"
    data-ecl-auto-init="ContentBlock"
    data-ecl-content-block=""
    data-test="data-test-value"
    data-test-1="data-test-value-1"
  >
    <ul
      class="ecl-content-block__label-container"
    >
      <li
        class="ecl-content-block__label-item"
      >
        <span
          class="ecl-label ecl-label--highlight"
        >
          highlight
        </span>
      </li>
      <li
        class="ecl-content-block__label-item"
      >
        <span
          class="ecl-label ecl-label--high"
        >
          high importance
        </span>
      </li>
    </ul>
    <ul
      class="ecl-content-block__primary-meta-container"
    >
      <li
        class="ecl-content-block__primary-meta-item"
      >
        PRIMARY META
      </li>
      <li
        class="ecl-content-block__primary-meta-item"
      >
        DD Month Year
      </li>
    </ul>
    <h1
      class="ecl-content-block__title"
      data-ecl-title-link=""
    >
      <a
        class="ecl-link ecl-link--standalone"
        href="/example"
      >
        Title
      </a>
    </h1>
    <div
      class="ecl-content-block__description"
    >
      Lorem ipsum dolor sit amet, consectetur adipiscing elit. Phasellus gravida ipsum ut lorem cursus, quis tincidunt sem viverra. Nunc vestibulum, mauris quis porta venenatis, justo odio commodo tellus
    </div>
    <ul
      class="ecl-content-block__secondary-meta-container"
    >
      <li
        class="ecl-content-block__secondary-meta-item"
      >
        <svg
          aria-hidden="true"
          class="ecl-icon ecl-icon--s ecl-content-block__secondary-meta-icon"
          focusable="false"
        >
          <use
            xlink:href="/icons.svg#calendar"
          />
        </svg>
        <span
          class="ecl-content-block__secondary-meta-label"
        >
          2018/10/22
        </span>
      </li>
      <li
        class="ecl-content-block__secondary-meta-item"
      >
        <svg
          aria-hidden="true"
          class="ecl-icon ecl-icon--s ecl-content-block__secondary-meta-icon"
          focusable="false"
        >
          <use
            xlink:href="/icons.svg#location"
          />
        </svg>
        <span
          class="ecl-content-block__secondary-meta-label"
        >
          Luxembourg
        </span>
      </li>
    </ul>
    <div
      class="ecl-content-block__list-container"
    >
      <dl
        class="ecl-description-list ecl-content-block__list"
      >
        <dt
          class="ecl-description-list__term"
        >
          Standalone links
        </dt>
        <dd
          class="ecl-description-list__definition ecl-description-list__definition--link"
        >
          <a
            class="ecl-link ecl-link--standalone ecl-description-list__definition-item"
            href="/example"
          >
            Lorem ipsum dolor sit amet
          </a>
          <a
            class="ecl-link ecl-link--standalone ecl-description-list__definition-item"
            href="/example"
          >
            Lorem ipsum dolor sit amet
          </a>
        </dd>
        <dt
          class="ecl-description-list__term"
        >
          Standard text
        </dt>
        <dd
          class="ecl-description-list__definition"
        >
          Lorem ipsum dolor sit amet, 
          <a
            class="ecl-link"
            href="/example"
          >
            consectetur adipiscing elit
          </a>
          .
        </dd>
        <dt
          class="ecl-description-list__term"
        >
          Links inline
        </dt>
        <dd
          class="ecl-description-list__definition ecl-description-list__definition--inline"
        >
          <a
            class="ecl-link ecl-link--standalone ecl-description-list__definition-item"
            href="/example"
          >
            Lorem ipsum dolor sit amet
          </a>
          <a
            class="ecl-link ecl-link--standalone ecl-description-list__definition-item"
            href="/example"
          >
            Lorem ipsum dolor sit amet
          </a>
        </dd>
        <dt
          class="ecl-description-list__term"
        >
          Taxonomy list
        </dt>
        <dd
          class="ecl-description-list__definition ecl-description-list__definition--taxonomy"
        >
          <span
            class="ecl-description-list__definition-item"
          >
            Taxonomy item 1
          </span>
          <span
            class="ecl-description-list__definition-item"
          >
            Taxonomy item 2
          </span>
          <span
            class="ecl-description-list__definition-item"
          >
            Taxonomy item 3
          </span>
        </dd>
      </dl>
    </div>
    <div
      class="ecl-content-block__link-container"
    >
      <ul
        class="ecl-content-block__link-list"
      >
        <li
          class="ecl-content-block__link-item"
        >
          <a
            class="ecl-link ecl-link--standalone"
            href="/example"
          >
            Primary link 1
          </a>
        </li>
        <li
          class="ecl-content-block__link-item"
        >
          <a
            class="ecl-link ecl-link--standalone"
            href="/example"
          >
            Primary link 2
          </a>
        </li>
        <li
          class="ecl-content-block__link-item"
        >
          <a
            class="ecl-link ecl-link--standalone"
            href="/example"
          >
            Primary link 3
          </a>
        </li>
        <li
          class="ecl-content-block__link-item"
        >
          <a
            class="ecl-link ecl-link--standalone"
            href="/example"
          >
            Primary link 4
          </a>
        </li>
      </ul>
      <ul
        class="ecl-content-block__link-list"
      >
        <li
          class="ecl-content-block__link-item"
        >
          <a
            class="ecl-link ecl-link--standalone"
            href="/example"
          >
            Secondary link 1
          </a>
        </li>
        <li
          class="ecl-content-block__link-item"
        >
          <a
            class="ecl-link ecl-link--standalone"
            href="/example"
          >
            Secondary link 2
          </a>
        </li>
      </ul>
    </div>
  </div>
</jest>
`;

exports[`Content block Default renders correctly with extra class names 1`] = `
<jest>
  <div
    class="ecl-content-block custom-class custom-class--test"
    data-ecl-auto-init="ContentBlock"
    data-ecl-content-block=""
  >
    <ul
      class="ecl-content-block__label-container"
    >
      <li
        class="ecl-content-block__label-item"
      >
        <span
          class="ecl-label ecl-label--highlight"
        >
          highlight
        </span>
      </li>
      <li
        class="ecl-content-block__label-item"
      >
        <span
          class="ecl-label ecl-label--high"
        >
          high importance
        </span>
      </li>
    </ul>
    <ul
      class="ecl-content-block__primary-meta-container"
    >
      <li
        class="ecl-content-block__primary-meta-item"
      >
        PRIMARY META
      </li>
      <li
        class="ecl-content-block__primary-meta-item"
      >
        DD Month Year
      </li>
    </ul>
    <h1
      class="ecl-content-block__title"
      data-ecl-title-link=""
    >
      <a
        class="ecl-link ecl-link--standalone"
        href="/example"
      >
        Title
      </a>
    </h1>
    <div
      class="ecl-content-block__description"
    >
      Lorem ipsum dolor sit amet, consectetur adipiscing elit. Phasellus gravida ipsum ut lorem cursus, quis tincidunt sem viverra. Nunc vestibulum, mauris quis porta venenatis, justo odio commodo tellus
    </div>
    <ul
      class="ecl-content-block__secondary-meta-container"
    >
      <li
        class="ecl-content-block__secondary-meta-item"
      >
        <svg
          aria-hidden="true"
          class="ecl-icon ecl-icon--s ecl-content-block__secondary-meta-icon"
          focusable="false"
        >
          <use
            xlink:href="/icons.svg#calendar"
          />
        </svg>
        <span
          class="ecl-content-block__secondary-meta-label"
        >
          2018/10/22
        </span>
      </li>
      <li
        class="ecl-content-block__secondary-meta-item"
      >
        <svg
          aria-hidden="true"
          class="ecl-icon ecl-icon--s ecl-content-block__secondary-meta-icon"
          focusable="false"
        >
          <use
            xlink:href="/icons.svg#location"
          />
        </svg>
        <span
          class="ecl-content-block__secondary-meta-label"
        >
          Luxembourg
        </span>
      </li>
    </ul>
    <div
      class="ecl-content-block__list-container"
    >
      <dl
        class="ecl-description-list ecl-content-block__list"
      >
        <dt
          class="ecl-description-list__term"
        >
          Standalone links
        </dt>
        <dd
          class="ecl-description-list__definition ecl-description-list__definition--link"
        >
          <a
            class="ecl-link ecl-link--standalone ecl-description-list__definition-item"
            href="/example"
          >
            Lorem ipsum dolor sit amet
          </a>
          <a
            class="ecl-link ecl-link--standalone ecl-description-list__definition-item"
            href="/example"
          >
            Lorem ipsum dolor sit amet
          </a>
        </dd>
        <dt
          class="ecl-description-list__term"
        >
          Standard text
        </dt>
        <dd
          class="ecl-description-list__definition"
        >
          Lorem ipsum dolor sit amet, 
          <a
            class="ecl-link"
            href="/example"
          >
            consectetur adipiscing elit
          </a>
          .
        </dd>
        <dt
          class="ecl-description-list__term"
        >
          Links inline
        </dt>
        <dd
          class="ecl-description-list__definition ecl-description-list__definition--inline"
        >
          <a
            class="ecl-link ecl-link--standalone ecl-description-list__definition-item"
            href="/example"
          >
            Lorem ipsum dolor sit amet
          </a>
          <a
            class="ecl-link ecl-link--standalone ecl-description-list__definition-item"
            href="/example"
          >
            Lorem ipsum dolor sit amet
          </a>
        </dd>
        <dt
          class="ecl-description-list__term"
        >
          Taxonomy list
        </dt>
        <dd
          class="ecl-description-list__definition ecl-description-list__definition--taxonomy"
        >
          <span
            class="ecl-description-list__definition-item"
          >
            Taxonomy item 1
          </span>
          <span
            class="ecl-description-list__definition-item"
          >
            Taxonomy item 2
          </span>
          <span
            class="ecl-description-list__definition-item"
          >
            Taxonomy item 3
          </span>
        </dd>
      </dl>
    </div>
    <div
      class="ecl-content-block__link-container"
    >
      <ul
        class="ecl-content-block__link-list"
      >
        <li
          class="ecl-content-block__link-item"
        >
          <a
            class="ecl-link ecl-link--standalone"
            href="/example"
          >
            Primary link 1
          </a>
        </li>
        <li
          class="ecl-content-block__link-item"
        >
          <a
            class="ecl-link ecl-link--standalone"
            href="/example"
          >
            Primary link 2
          </a>
        </li>
        <li
          class="ecl-content-block__link-item"
        >
          <a
            class="ecl-link ecl-link--standalone"
            href="/example"
          >
            Primary link 3
          </a>
        </li>
        <li
          class="ecl-content-block__link-item"
        >
          <a
            class="ecl-link ecl-link--standalone"
            href="/example"
          >
            Primary link 4
          </a>
        </li>
      </ul>
      <ul
        class="ecl-content-block__link-list"
      >
        <li
          class="ecl-content-block__link-item"
        >
          <a
            class="ecl-link ecl-link--standalone"
            href="/example"
          >
            Secondary link 1
          </a>
        </li>
        <li
          class="ecl-content-block__link-item"
        >
          <a
            class="ecl-link ecl-link--standalone"
            href="/example"
          >
            Secondary link 2
          </a>
        </li>
      </ul>
    </div>
  </div>
</jest>
`;

exports[`Content block Default renders correctly with list extra classes 1`] = `
<jest>
  <div
    class="ecl-content-block"
    data-ecl-auto-init="ContentBlock"
    data-ecl-content-block=""
  >
    <ul
      class="ecl-content-block__label-container"
    >
      <li
        class="ecl-content-block__label-item"
      >
        <span
          class="ecl-label ecl-label--highlight"
        >
          highlight
        </span>
      </li>
      <li
        class="ecl-content-block__label-item"
      >
        <span
          class="ecl-label ecl-label--high"
        >
          high importance
        </span>
      </li>
    </ul>
    <ul
      class="ecl-content-block__primary-meta-container"
    >
      <li
        class="ecl-content-block__primary-meta-item"
      >
        PRIMARY META
      </li>
      <li
        class="ecl-content-block__primary-meta-item"
      >
        DD Month Year
      </li>
    </ul>
    <h1
      class="ecl-content-block__title"
      data-ecl-title-link=""
    >
      <a
        class="ecl-link ecl-link--standalone"
        href="/example"
      >
        Title
      </a>
    </h1>
    <div
      class="ecl-content-block__description"
    >
      Lorem ipsum dolor sit amet, consectetur adipiscing elit. Phasellus gravida ipsum ut lorem cursus, quis tincidunt sem viverra. Nunc vestibulum, mauris quis porta venenatis, justo odio commodo tellus
    </div>
    <ul
      class="ecl-content-block__secondary-meta-container"
    >
      <li
        class="ecl-content-block__secondary-meta-item"
      >
        <svg
          aria-hidden="true"
          class="ecl-icon ecl-icon--s ecl-content-block__secondary-meta-icon"
          focusable="false"
        >
          <use
            xlink:href="/icons.svg#calendar"
          />
        </svg>
        <span
          class="ecl-content-block__secondary-meta-label"
        >
          2018/10/22
        </span>
      </li>
      <li
        class="ecl-content-block__secondary-meta-item"
      >
        <svg
          aria-hidden="true"
          class="ecl-icon ecl-icon--s ecl-content-block__secondary-meta-icon"
          focusable="false"
        >
          <use
            xlink:href="/icons.svg#location"
          />
        </svg>
        <span
          class="ecl-content-block__secondary-meta-label"
        >
          Luxembourg
        </span>
      </li>
    </ul>
    <div
      class="ecl-content-block__list-container"
    >
      <dl
        class="ecl-description-list ecl-content-block__list a-list-extra-class"
      >
        <dt
          class="ecl-description-list__term"
        >
          Standalone links
        </dt>
        <dd
          class="ecl-description-list__definition ecl-description-list__definition--link"
        >
          <a
            class="ecl-link ecl-link--standalone ecl-description-list__definition-item"
            href="/example"
          >
            Lorem ipsum dolor sit amet
          </a>
          <a
            class="ecl-link ecl-link--standalone ecl-description-list__definition-item"
            href="/example"
          >
            Lorem ipsum dolor sit amet
          </a>
        </dd>
        <dt
          class="ecl-description-list__term"
        >
          Standard text
        </dt>
        <dd
          class="ecl-description-list__definition"
        >
          Lorem ipsum dolor sit amet, 
          <a
            class="ecl-link"
            href="/example"
          >
            consectetur adipiscing elit
          </a>
          .
        </dd>
        <dt
          class="ecl-description-list__term"
        >
          Links inline
        </dt>
        <dd
          class="ecl-description-list__definition ecl-description-list__definition--inline"
        >
          <a
            class="ecl-link ecl-link--standalone ecl-description-list__definition-item"
            href="/example"
          >
            Lorem ipsum dolor sit amet
          </a>
          <a
            class="ecl-link ecl-link--standalone ecl-description-list__definition-item"
            href="/example"
          >
            Lorem ipsum dolor sit amet
          </a>
        </dd>
        <dt
          class="ecl-description-list__term"
        >
          Taxonomy list
        </dt>
        <dd
          class="ecl-description-list__definition ecl-description-list__definition--taxonomy"
        >
          <span
            class="ecl-description-list__definition-item"
          >
            Taxonomy item 1
          </span>
          <span
            class="ecl-description-list__definition-item"
          >
            Taxonomy item 2
          </span>
          <span
            class="ecl-description-list__definition-item"
          >
            Taxonomy item 3
          </span>
        </dd>
      </dl>
    </div>
    <div
      class="ecl-content-block__link-container"
    >
      <ul
        class="ecl-content-block__link-list"
      >
        <li
          class="ecl-content-block__link-item"
        >
          <a
            class="ecl-link ecl-link--standalone"
            href="/example"
          >
            Primary link 1
          </a>
        </li>
        <li
          class="ecl-content-block__link-item"
        >
          <a
            class="ecl-link ecl-link--standalone"
            href="/example"
          >
            Primary link 2
          </a>
        </li>
        <li
          class="ecl-content-block__link-item"
        >
          <a
            class="ecl-link ecl-link--standalone"
            href="/example"
          >
            Primary link 3
          </a>
        </li>
        <li
          class="ecl-content-block__link-item"
        >
          <a
            class="ecl-link ecl-link--standalone"
            href="/example"
          >
            Primary link 4
          </a>
        </li>
      </ul>
      <ul
        class="ecl-content-block__link-list"
      >
        <li
          class="ecl-content-block__link-item"
        >
          <a
            class="ecl-link ecl-link--standalone"
            href="/example"
          >
            Secondary link 1
          </a>
        </li>
        <li
          class="ecl-content-block__link-item"
        >
          <a
            class="ecl-link ecl-link--standalone"
            href="/example"
          >
            Secondary link 2
          </a>
        </li>
      </ul>
    </div>
  </div>
</jest>
`;

exports[`Content block Default renders correctly with simple title 1`] = `
<jest>
  <div
    class="ecl-content-block"
    data-ecl-auto-init="ContentBlock"
    data-ecl-content-block=""
  >
    <ul
      class="ecl-content-block__label-container"
    >
      <li
        class="ecl-content-block__label-item"
      >
        <span
          class="ecl-label ecl-label--highlight"
        >
          highlight
        </span>
      </li>
      <li
        class="ecl-content-block__label-item"
      >
        <span
          class="ecl-label ecl-label--high"
        >
          high importance
        </span>
      </li>
    </ul>
    <ul
      class="ecl-content-block__primary-meta-container"
    >
      <li
        class="ecl-content-block__primary-meta-item"
      >
        PRIMARY META
      </li>
      <li
        class="ecl-content-block__primary-meta-item"
      >
        DD Month Year
      </li>
    </ul>
    <h1
      class="ecl-content-block__title"
    >
      title
    </h1>
    <div
      class="ecl-content-block__description"
    >
      Lorem ipsum dolor sit amet, consectetur adipiscing elit. Phasellus gravida ipsum ut lorem cursus, quis tincidunt sem viverra. Nunc vestibulum, mauris quis porta venenatis, justo odio commodo tellus
    </div>
    <ul
      class="ecl-content-block__secondary-meta-container"
    >
      <li
        class="ecl-content-block__secondary-meta-item"
      >
        <svg
          aria-hidden="true"
          class="ecl-icon ecl-icon--s ecl-content-block__secondary-meta-icon"
          focusable="false"
        >
          <use
            xlink:href="/icons.svg#calendar"
          />
        </svg>
        <span
          class="ecl-content-block__secondary-meta-label"
        >
          2018/10/22
        </span>
      </li>
      <li
        class="ecl-content-block__secondary-meta-item"
      >
        <svg
          aria-hidden="true"
          class="ecl-icon ecl-icon--s ecl-content-block__secondary-meta-icon"
          focusable="false"
        >
          <use
            xlink:href="/icons.svg#location"
          />
        </svg>
        <span
          class="ecl-content-block__secondary-meta-label"
        >
          Luxembourg
        </span>
      </li>
    </ul>
    <div
      class="ecl-content-block__list-container"
    >
      <dl
        class="ecl-description-list ecl-content-block__list"
      >
        <dt
          class="ecl-description-list__term"
        >
          Standalone links
        </dt>
        <dd
          class="ecl-description-list__definition ecl-description-list__definition--link"
        >
          <a
            class="ecl-link ecl-link--standalone ecl-description-list__definition-item"
            href="/example"
          >
            Lorem ipsum dolor sit amet
          </a>
          <a
            class="ecl-link ecl-link--standalone ecl-description-list__definition-item"
            href="/example"
          >
            Lorem ipsum dolor sit amet
          </a>
        </dd>
        <dt
          class="ecl-description-list__term"
        >
          Standard text
        </dt>
        <dd
          class="ecl-description-list__definition"
        >
          Lorem ipsum dolor sit amet, 
          <a
            class="ecl-link"
            href="/example"
          >
            consectetur adipiscing elit
          </a>
          .
        </dd>
        <dt
          class="ecl-description-list__term"
        >
          Links inline
        </dt>
        <dd
          class="ecl-description-list__definition ecl-description-list__definition--inline"
        >
          <a
            class="ecl-link ecl-link--standalone ecl-description-list__definition-item"
            href="/example"
          >
            Lorem ipsum dolor sit amet
          </a>
          <a
            class="ecl-link ecl-link--standalone ecl-description-list__definition-item"
            href="/example"
          >
            Lorem ipsum dolor sit amet
          </a>
        </dd>
        <dt
          class="ecl-description-list__term"
        >
          Taxonomy list
        </dt>
        <dd
          class="ecl-description-list__definition ecl-description-list__definition--taxonomy"
        >
          <span
            class="ecl-description-list__definition-item"
          >
            Taxonomy item 1
          </span>
          <span
            class="ecl-description-list__definition-item"
          >
            Taxonomy item 2
          </span>
          <span
            class="ecl-description-list__definition-item"
          >
            Taxonomy item 3
          </span>
        </dd>
      </dl>
    </div>
    <div
      class="ecl-content-block__link-container"
    >
      <ul
        class="ecl-content-block__link-list"
      >
        <li
          class="ecl-content-block__link-item"
        >
          <a
            class="ecl-link ecl-link--standalone"
            href="/example"
          >
            Primary link 1
          </a>
        </li>
        <li
          class="ecl-content-block__link-item"
        >
          <a
            class="ecl-link ecl-link--standalone"
            href="/example"
          >
            Primary link 2
          </a>
        </li>
        <li
          class="ecl-content-block__link-item"
        >
          <a
            class="ecl-link ecl-link--standalone"
            href="/example"
          >
            Primary link 3
          </a>
        </li>
        <li
          class="ecl-content-block__link-item"
        >
          <a
            class="ecl-link ecl-link--standalone"
            href="/example"
          >
            Primary link 4
          </a>
        </li>
      </ul>
      <ul
        class="ecl-content-block__link-list"
      >
        <li
          class="ecl-content-block__link-item"
        >
          <a
            class="ecl-link ecl-link--standalone"
            href="/example"
          >
            Secondary link 1
          </a>
        </li>
        <li
          class="ecl-content-block__link-item"
        >
          <a
            class="ecl-link ecl-link--standalone"
            href="/example"
          >
            Secondary link 2
          </a>
        </li>
      </ul>
    </div>
  </div>
</jest>
`;
